core/src/main/java/de/danoeh/antennapod/core/storage/DBWriter.java

```diff
@@ -118,30 +126,59 @@ private static boolean deleteFeedMediaSynchronous(
                 EventBus.getDefault().post(evt);
                 return false;
             }
-            media.setDownloaded(false);
-            media.setFile_url(null);
-            media.setHasEmbeddedPicture(false);
-            PodDBAdapter adapter = PodDBAdapter.getInstance();
-            adapter.open();
-            adapter.setMedia(media);
-            adapter.close();
+        } else if (media.getFile_url().startsWith("content://")) {
+            // Local feed
 
-            if (media.getId() == PlaybackPreferences.getCurrentlyPlayingFeedMediaId()) {
-                PlaybackPreferences.writeNoMediaPlaying();
-                IntentUtils.sendLocalBroadcast(context, PlaybackServiceInterface.ACTION_SHUTDOWN_PLAYBACK_SERVICE);
+            DocumentFile documentFile = DocumentFile.fromSingleUri(
+                    context,
+                    Uri.parse(media.getFile_url())
+            );
 
-                NotificationManagerCompat nm = NotificationManagerCompat.from(context);
-                nm.cancel(R.id.notification_playing);
+            if (documentFile == null || !documentFile.exists() || !documentFile.delete()) {
+                MessageEvent evt = new MessageEvent(context.getString(R.string.delete_local_failed));
+                EventBus.getDefault().post(evt);
+                return false;
             }
 
-            // Gpodder: queue delete action for synchronization
-            FeedItem item = media.getItem();
-            EpisodeAction action = new EpisodeAction.Builder(item, EpisodeAction.DELETE)
-                    .currentTimestamp()
-                    .build();
-            SynchronizationQueueSink.enqueueEpisodeActionIfSynchronizationIsActive(context, action);
+            localDelete = true;
+        }
+
+        media.setDownloaded(false);
+        media.setFile_url(null);
+        media.setHasEmbeddedPicture(false);
+        PodDBAdapter adapter = PodDBAdapter.getInstance();
+        adapter.open();
+        adapter.setMedia(media);
+        adapter.close();
+
+        if (media.getId() == PlaybackPreferences.getCurrentlyPlayingFeedMediaId()) {
+            PlaybackPreferences.writeNoMediaPlaying();
+            IntentUtils.sendLocalBroadcast(context, PlaybackServiceInterface.ACTION_SHUTDOWN_PLAYBACK_SERVICE);
+
+            NotificationManagerCompat nm = NotificationManagerCompat.from(context);
+            nm.cancel(R.id.notification_playing);
+        }
+
+        // Gpodder: queue delete action for synchronization
+        FeedItem item = media.getItem();
+        EpisodeAction action = new EpisodeAction.Builder(item, EpisodeAction.DELETE)
+                .currentTimestamp()
+                .build();
+        SynchronizationQueueSink.enqueueEpisodeActionIfSynchronizationIsActive(context, action);
```

Comment:
Now this is queued even for local files, which should not be synced at all

Reply:
Thanks, updated